src/olympia/users/admin.py

```diff
@@ -171,37 +171,69 @@ def get_list_display(self, request):
         search_term = (
             search_form.cleaned_data.get(SEARCH_VAR) if search_form.is_valid() else None
         )
-        if search_term and self.ip_addresses_if_query_is_all_ip_addresses(search_term):
+        if search_term and self.ip_addresses_and_networks_from_query(search_term):
             return (*self.list_display, *self.extra_list_display_for_ip_searches)
         return self.list_display
 
-    def ip_addresses_if_query_is_all_ip_addresses(self, search_term):
+    def ip_addresses_and_networks_from_query(self, search_term):
         # Caller should already have cleaned up search_term at this point,
         # removing whitespace etc if there is a comma separating multiple
         # terms.
         search_terms = search_term.split(',')
         ips = []
+        networks = []
         for term in search_terms:
+            # If term is a number, skip trying to recognize an IP address
+            # entirely, because ip_address() is able to understand IP addresses
+            # as integers, and we don't want that, it's likely an user ID.
+            if term.isdigit():
+                return None
+            # Is the search term an IP ?
             try:
-                ip = ipaddress.ip_address(term)
-                ip_str = str(ip)
-                if ip_str != term:
-                    raise ValueError
-                ips.append(ip_str)
+                ips.append(ipaddress.ip_address(term))
+                continue
             except ValueError:
-                break
-        if search_terms == ips:
-            # If all we are searching for are IPs, we'll use our custom IP
-            # search.
-            # Note that this comparison relies on ips being stored as strings,
-            # if that were to change that it would break.
-            return ips
-        return None
+                pass
+            # Is the search term a network ?
+            try:
+                networks.append(ipaddress.ip_network(term))
+                continue
+            except ValueError:
+                pass
+            # Is the search term an IP range ?
+            if term.count('-') == 1:
+                try:
+                    networks.extend(
+                        ipaddress.summarize_address_range(
+                            *(ipaddress.ip_address(i.strip()) for i in term.split('-'))
```

Comment:
... I can't help thinking there is a way to write this that doesn't need all the iterator plus expansion magic for exactly 2 args but this works...